_posts/2023-07-17-elytron-oidc-client-scope.adoc

```diff
@@ -0,0 +1,153 @@
+
+toc::[]
+
+Note: This quickstart demonstrates how to configure scope values in two different ways: 
+
+* Deployment Configuration
+* Subsystem Configuration
+
+== Setting up Keycloak OpenID Provider
+
+It’s easy to set up Keycloak using Docker. Follow the steps in Keycloak’s https://www.keycloak.org/getting-started/getting-started-docker[ getting started guide ] to start Keycloak, create a realm called `myrealm` and register a client called `myclient`.
+
+After registering our client, `myclient`, we also need to configure valid redirect URIs. Simply click on `Clients` and then on `myclient`. In the `Valid Redirect URIs field`, enter http://localhost:8090/simple-webapp-oidc/*.
+
+Navigate to the `client scopes` tab for `myclient` and change the `assigned-type` for all scope values from `default` to `optional`. This ensures that you will only get access to claims that you request using the scope values you configure. 
+
+Now, click on Realm roles and create two roles, user and admin. 
+
+Finally, create a user called `alice` whose `First name` is Alice and `Last name` is Smith and assign her the user and admin roles. Steps for assigning roles can be found in the https://www.keycloak.org/docs/latest/server_admin/#proc-assigning-role-mappings_server_administration_guide[ Keycloak documentation]. We will be using different scopes to query these pieces of information about Alice. 
+
+Next, navigate to the `credentials` tab and create a password for Alice. Toggle the `temporary` switch off so you are not prompted to update the password after the first login. The list of scope are the values Keycloak allows and it varies for different OpenID providers.
+
+
+Note: The realm and client names are case sensitive. 
+
+== Server Configuration
+
+First clone the `elytron-example` repo locally: 
+```
+    git clone https://github.com/wildfly-security-incubator/elytron-examples
+    cd elytron-examples
+```
+
+In order to authenticate a user using OpenID connect, we need to configure the server to use OpenID attributes. This can be done using one of the following configurations: 
+
+=== Deployment Configuration
+
+The configuration required to secure an application with OpenID Connect can be specified in the deployment using the `oidc.json` file.
+
+The first step is to create an `oidc.json` configuration file in the `WEB-INF` directory of the `simple-webapp-oidc` project. 
+
+Note: There is a `simple-webapp-oidc` project inside the `elytron-examples` directory. However, we will not be using that one. Instead, we will be using the slightly modified version of that inside the `elytron-oidc-client-scope` project. 
+
+There is already an `oidc.json` file inside the `WEB-INF` directory. So, we will only be editing that file by adding scope values to the existing deployment configuration. The following is the contents of the `oidc.json` file used in this example:
```

Comment:
This sentence is a bit confusing because it says we need to edit the file but the scope value is already present in the file.